src/Sets/Tetrahedron.jl

```diff
@@ -0,0 +1,111 @@
+export Tetrahedron
+
+"""
+    Tetrahedron{N, VN<:AbstractVector{N}, VT<:AbstractVector{VN}} <: AbstractPolytope{N}
+
+Type that represents a tetrahedron in vertex representation.
+
+### Fields
+
+- `vertices` -- list of vertices
+
+### Examples
+
+A tetrahedron can be constructed by passing the list of vertices.
+The following builds the tetrahedron with edge length 2 from the [wikipedia page Tetrahedron](https://en.wikipedia.org/wiki/Tetrahedron): 
+
+```jldoctest
+julia> vertices = [[1, 0, -1/sqrt(2)], [-1, 0, -1/sqrt(2)], [0, 1, 1/sqrt(2)], [0, -1, 1/sqrt(2)]];
+
+julia> T = Tetrahedron(vertices); T isa Tetrahedron
+true
+
+julia> zeros(3) ∈ T
+true
+
+julia> dim(T)
+3
+
+julia> σ(ones(3), T)
+3-element Vector{Float64}:
+ 0.0
+ 1.0
+ 0.7071067811865475
+```
+"""
+struct Tetrahedron{N,VN<:AbstractVector{N},VT<:AbstractVector{VN}} <: AbstractPolytope{N}
+    vertices::VT
+end
+
+isoperationtype(::Type{<:Tetrahedron}) = false
+
+# constructor with empty vertices list
+Tetrahedron{N}() where {N} = Tetrahedron(Vector{Vector{N}}())
+
+# constructor with no vertices of type Float64
+Tetrahedron() = Tetrahedron{Float64}()
+
+# constructor from rectangular matrix
+function Tetrahedron(vertices_matrix::MT) where {N,MT<:AbstractMatrix{N}}
+    vertices = [vertices_matrix[:, j] for j in axes(vertices_matrix, 2)]
```

Comment:
```suggestion
    vertices = [c for c in eachcol(vertices_matrix)]
```

```julia
julia> M = rand(2, 2)
2×2 Matrix{Float64}:
 0.118181  0.549152
 0.925909  0.79909

julia> @time [M[:, j] for j in axes(M, 2)]
  0.034929 seconds (43.36 k allocations: 2.315 MiB, 99.19% compilation time)
 [0.11818109598081683, 0.9259090178191179]
 [0.5491517086373754, 0.7990903995880422]

julia> @time [c for c in eachcol(M)]
  0.000011 seconds (3 allocations: 208 bytes)
2-element Vector{SubArray{Float64, 1, Matrix{Float64}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}:
 [0.11818109598081683, 0.9259090178191179]
 [0.5491517086373754, 0.7990903995880422]
```

Reply:
that's due to measuring in global scope.

```julia
function Tetrahedron(vertices_matrix::MT) where {N,MT<:AbstractMatrix{N}}
    vertices = [vertices_matrix[:, j] for j in axes(vertices_matrix, 2)]
    return Tetrahedron(vertices)
end
```

```julia
julia> M = rand(3, 4);

julia> @btime LazySets.Tetrahedron($M);
  146.802 ns (5 allocations: 400 bytes)
```

though

```julia
function Tetrahedron2(vertices_matrix::MT) where {N,MT<:AbstractMatrix{N}}
    vertices = collect(eachcol(vertices_matrix))
    return Tetrahedron(vertices)
end
```

```julia
julia> @btime LazySets.Tetrahedron2($M);
  36.170 ns (1 allocation: 224 bytes)
```

Reply:
the difference is due to `vertices_matrix[:, j]` allocates while `eachcol(..)` returns views